homeassistant/helpers/selector.py

```diff
@@ -74,44 +74,53 @@ def serialize(self) -> Any:
         return {"selector": {self.selector_type: self.config}}
 
 
+SINGLE_ENTITY_SELECTOR_CONFIG_SCHEMA = vol.Schema(
+    {
+        # Integration that provided the entity
+        vol.Optional("integration"): str,
+        # Domain the entity belongs to
+        vol.Optional("domain"): str,
+        # Device class of the entity
+        vol.Optional("device_class"): str,
+    }
+)
+
+
 @SELECTORS.register("entity")
 class EntitySelector(Selector):
-    """Selector of a single entity."""
+    """Selector of a single or list of entities."""
 
     selector_type = "entity"
 
-    CONFIG_SCHEMA = vol.Schema(
-        {
-            # Integration that provided the entity
-            vol.Optional("integration"): str,
-            # Domain the entity belongs to
-            vol.Optional("domain"): str,
-            # Device class of the entity
-            vol.Optional("device_class"): str,
-        }
+    CONFIG_SCHEMA = SINGLE_ENTITY_SELECTOR_CONFIG_SCHEMA.extend(
+        {vol.Optional("multiple", default=False): cv.boolean}
     )
 
-    def __call__(self, data: Any) -> str:
+    def __call__(self, data: Any) -> str | list[str]:
         """Validate the passed selection."""
-        try:
-            entity_id = cv.entity_id(data)
-            domain = split_entity_id(entity_id)[0]
-        except vol.Invalid:
-            # Not a valid entity_id, maybe it's an entity entry id
-            return cv.entity_id_or_uuid(cv.string(data))
-        else:
-            if "domain" in self.config and domain != self.config["domain"]:
-                raise vol.Invalid(
-                    f"Entity {entity_id} belongs to domain {domain}, "
-                    f"expected {self.config['domain']}"
-                )
-
-            return entity_id
+
+        def validate(e_or_u: str) -> str:
+            if not valid_entity_id(e_or_u):
+                return e_or_u
+            if allowed_domain := self.config.get("domain"):
+                domain = split_entity_id(e_or_u)[0]
+                if domain != allowed_domain:
+                    raise vol.Invalid(
+                        f"Entity {e_or_u} belongs to domain {domain}, "
+                        f"expected {allowed_domain}"
+                    )
+            return e_or_u
+
+        if not self.config["multiple"]:
+            return validate(cv.entity_id_or_uuid(cv.string(data)))
```

Comment:
Looks like you could move `cv.entity_id_or_uuid(cv.string(data))` inside `validate` too ? 

Is `cv.string` necessary on top of `cv.entity_id_or_uuid` ?

Reply:
Fixed 👍